### ISSUE TITLE
[Docs] docs for IAM, service accounts, labels and annotations

### ISSUE THREAD

**Issue by kumare3** [flyteorg/flyte#1512]
**Describe the documentation update**
 There is no good reference explaining how to set service accounts, roles, labels, and annotations.

**GitHub repo(s)**
[flytesnacks](https://github.com/flyteorg/flytesnacks)

**Comment by samhita-alla** [comment]
@kumare3 We have a [page](https://docs.flyte.org/projects/cookbook/en/latest/auto/core/containerization/workflow_labels_annotations.html#sphx-glr-auto-core-containerization-workflow-labels-annotations-py) dedicated to labels and annotations. Will that not suffice?

Also, where should service accounts and roles be documented? Do we have any references that can help in writing down the doc?

**Comment by vikramadityasinghs** [comment]
could I be assigned to this? @kumare3  @samhita-alla

**Comment by kumare3** [comment]
@samhita-alla yes when you create a launch plan, these values are all available to be populated as shown in launchplan spec - https://docs.flyte.org/projects/flyteidl/en/stable/protos/docs/admin/admin.html#launchplanspec

**Comment by samhita-alla** [comment]
> could I be assigned to this? @kumare3 @samhita-alla

@vikramadityasinghs sure, please go ahead! Let us know if you have questions. If you'd like to join our slack, here's the [link](https://docs.google.com/forms/d/e/1FAIpQLScWPJZncyL-bBCWbuQ9HWuMzA8-r2RJc__CSWJoAYUQADN-BQ/viewform).

**Comment by cosmicBboy** [comment]
TODO: still need to document how to set service accounts
